openedx/core/djangoapps/content/learning_sequences/api/tests/test_outlines.py

```diff
@@ -371,6 +374,101 @@ def test_is_due_and_after_due(self):
+
+        # The UsageKeys we're going to set up for date tests.
+        cls.section_key = cls.course_key.make_usage_key('chapter', 'ch1')
+
+        # Sequence with due date
+        cls.seq_due_key = cls.course_key.make_usage_key('sequential', 'seq')
+
+        # Sequence with due date and "inaccessible after due" enabled
+        cls.seq_hide_after_due_key = cls.course_key.make_usage_key('sequential', 'seq_hide_after_due_key')
+
+        # Set scheduling information into edx-when for a single Section with
+        # two sequences with due date
+        set_dates_for_course(
+            cls.course_key,
+            [
+                (
+                    cls.course_key.make_usage_key('course', 'course'),
+                    {
+                        'start': datetime(2020, 5, 10, tzinfo=timezone.utc),
+                    }
+                ),
+                (
+                    cls.section_key,
+                    {'start': datetime(2020, 5, 15, tzinfo=timezone.utc)}
+                ),
+                (
+                    cls.seq_due_key,
+                    {'due': datetime(2020, 5, 21, tzinfo=timezone.utc)}
+                ),
+                (
+                    cls.seq_hide_after_due_key,
+                    {'due': datetime(2020, 5, 21, tzinfo=timezone.utc)}
+                ),
+            ]
+        )
+        visibility = VisibilityData(
+            hide_from_toc=False,
+            visible_to_staff_only=False
+        )
+        cls.outline = CourseOutlineData(
+            course_key=cls.course_key,
+            title="User Outline Test Course!",
+            published_at=datetime(2020, 5, 20, tzinfo=timezone.utc),
+            published_version="5ebece4b69dd593d82fe2020",
+            sections=[
+                CourseSectionData(
+                    usage_key=cls.section_key,
+                    title="Section",
+                    visibility=visibility,
+                    sequences=[
+                        CourseLearningSequenceData(
+                            usage_key=cls.seq_due_key,
+                            title='Due',
+                            visibility=visibility
+                        ),
+                        CourseLearningSequenceData(
+                            usage_key=cls.seq_hide_after_due_key,
+                            title='Inaccessible after due',
+                            visibility=visibility,
+                            inaccessible_after_due=True
+                        ),
+                    ],
+                ),
+            ],
+            self_paced=True,
+        )
+
+        replace_course_outline(cls.outline)
+
+    def test_sequences_accessible_after_due(self):
+        at_time = datetime(2020, 5, 22, tzinfo=timezone.utc)
+        staff_details = get_user_course_outline_details(self.course_key, self.global_staff, at_time)
+        student_details = get_user_course_outline_details(self.course_key, self.student, at_time)
```

Comment:
@0x29a, nit: can these be named `staff_outline`, `student_outline`?

Reply:
@lgp171188, `get_user_course_outline_details` returns not only outline, but also schedule details.
Also, there are many other tests using this pattern.

Probably this can be refactored to be like:
```python
staff_outline = get_user_course_outline_details(
    self.course_key, self.global_staff, at_time
).outline

student_outline = get_user_course_outline_details(
    self.course_key, self.student, at_time
).outline
```

But I intend to keep the diff as small as possible.

Anyway, I can include this change. What do you think?

Reply:
@0x29a, since we are only using the `outline` property, I'm okay with the refactoring 👍🏼